config/initializers/okcomputer.rb

```diff
@@ -35,31 +35,43 @@ def table_check(klass)
 end
 OkComputer::Registry.register 'feature-tables-have-data', TablesHaveDataCheck.new
 
+# rubocop:disable Metrics/AbcSize
 # check that directory is accessible without consideration for writability
 class DirectoryExistsCheck < OkComputer::Check
-  attr_accessor :directory
+  attr_accessor :directory, :min_subfolder_count
 
-  def initialize(directory)
+  def initialize(directory, min_subfolder_count = nil)
     self.directory = directory
+    self.min_subfolder_count = min_subfolder_count
   end
 
   def check
-    stat = File.stat(directory) if File.exist?(directory)
-    if stat
-      if stat.directory?
-        mark_message "'#{directory}' is a reachable directory"
-      else
-        mark_message "'#{directory}' is not a directory."
-        mark_failure
-      end
-    else
+    unless File.exist? directory
       mark_message "Directory '#{directory}' does not exist."
       mark_failure
     end
+
+    unless File.directory? directory
+      mark_message "'#{directory}' is not a directory."
+      mark_failure
+    end
+
+    mark_message "'#{directory}' is a reachable directory"
+    if min_subfolder_count && Dir.entries(directory).size > min_subfolder_count
+      mark_message "'#{directory}' has the required minimum number of subfolders (#{min_subfolder_count})"
+    elsif min_subfolder_count
+      mark_message "'#{directory}' does not have the required minimum number of subfolders (#{min_subfolder_count})"
+      mark_failure
+    end
   end
 end
+# rubocop:enable Metrics/AbcSize
+
 Settings.storage_root_map.default.each do |name, location|
-  OkComputer::Registry.register "feature-#{name}", DirectoryExistsCheck.new(location)
+  deposit_location = "#{location}/deposit"
+  sdrobjects_locations = "#{location}/sdr2objects"
```

Comment:
unlikely to make a material difference in the near future, and will alert loudly/harmlessly if things do change... but just for consistency, would you mind doing:
```ruby
sdrobjects_locations = "#{location}/#{Settings.moab.storage_trunk}"
```

i was going to suggest similar for the `deposit_location` init, but it turns out we don't configure that in this app, AFAICT, since pres cat never directly deals with deposits.  i dunno, maybe that indicates that we should just get rid of the deposit check part (but otherwise keep the logic in this PR the same).  or maybe it indicates this comment is too nitpicky.  your call on whether to change this 🙂

Reply:
if pres_cat doesn't care about the "deposits" directory, then I would argue the okcomputer check herer shouldn't care either  - and I have removed that explicit check

Reply:
makes sense, apologies for the misleading suggestion in the ticket